src/dra/42398.c

```diff
@@ -949,8 +949,33 @@ s32 func_800E9640(s32 arg0, s32 arg1, s32 arg2, s32* readBufferAddress,
     return ret;
 }
 
-// https://decomp.me/scratch/6dWJG very close (branching)
-INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800E96E8);
+s32 func_800E96E8(s32 arg0, s32 arg1, s32 arg2, void* arg3, s32 arg4,
+                  s32 arg5) {
+    s8 sp18[32];
```

Comment:
I kind-of recognise this function. It looks like it's creating a string with the path of where the save file is actually found. You can rename `sp18` as `buffer` or `filePath` or `savePath` or whatever name you find more suitable